src/utils.js

```diff
@@ -228,3 +229,35 @@ export async function hashPassword(password) {
     return Promise.reject(new Error("Password wasn't provided"))
   }
 }
+
+export const checkUserPermission = async (ctx, operation, permission, permissionSpecified, resource) => {
```

Comment:
Maybe we could find a way to move the *-specified check into this as well as the check are all very similar in the API endpoints at the moment. It would help to centralise that.

Reply:
it already does. The permissionSpecified is '*-specified' . Example usage of this function - `checkUserPermission(ctx, 'removeChannel', 'channel-manage-all', 'channel-manage-specified', channelId)`

Reply:
Ok, I understand now.